### PR TITLE
Run SAST checks in CI

### PR DESCRIPTION
**Pull request by matmair**
We are currently failing the SAST section of https://securityscorecards.dev/viewer/?uri=github.com/inventree/InvenTree because it seems like SonarCloud is not running on (all?) PRs. The easiest solution for this is to just run it in CI, this should also help to discover problematic code sooner.

This PR:
- adds a workflow to run sonar's checks on master and pull requests
- adds the minimal config required for running sonar in CI

### CODE REVIEW

**Comment by matmair** [comment]
This will require the enviroment variable `SONAR_TOKEN ` to be set - will be provided on another channel

File: .github/workflows/sast.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (wolflu05) [inline .github/workflows/sast.yaml:20]: Q: when should this workflow execute? Are you aware, that this secret is never available with the pull_request trigger, due to protection mechanisms?

Comment (matmair) [reply .github/workflows/sast.yaml:20]: This is 1:1 from the docs of SonarCloud, I trust that they validated this approach
https://github.com/SonarSource/sonarcloud-github-action?tab=readme-ov-file#usage

Comment (wolflu05) [reply .github/workflows/sast.yaml:20]: Ok, interesting, [GitHub itself](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/#:~:text=Due%20to%20the%20dangers%20inherent%20to%20automatic%20processing%20of%20PRs%2C%20GitHub%E2%80%99s%20standard%20pull_request%20workflow%20trigger%20by%20default%20prevents%20write%20permissions%20and%20secrets%20access%20to%20the%20target%20repository.) tells something else, but happy to have learned something new.

**Comment by SchrodingersGat** [comment]
@matmair `SONAR_TOKEN` has now been added

**Comment by matmair** [comment]
Sonar / @agigleux have fixed the underlying issue, this is not needed anymore.

@SchrodingersGat you can remove the secret `SONAR_TOKEN`, it will not be needed

**Comment by SchrodingersGat** [comment]
Awesome, thanks @agigleux and Sonar!

**Pull request closed without merging**